### PR TITLE
Replace unwrap in prod code with expect

### PR DESCRIPTION
**Pull request by THenry14**
Closes https://github.com/foundry-rs/starknet-foundry/issues/1346

## Introduced changes

- changed unwrap to expect

## Checklist

- [X] Linked relevant issue
- [X] Updated relevant documentation
- [X] Added relevant tests
- [X] Performed self-review of the code
- [X] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by tomek0123456789 — APPROVED** [review]

File: crates/sncast/tests/data/multicall_configs/deploy_invoke_calldata_ids.toml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 call_type = "invoke"
-contract_address = "0x50E0FDC4FEEB0CB85F175D40EC2F8D622A92CCAC3E59419C773B4EAAD2B5A2"
+contract_address = "0x50e0fdc4feeb0cb85f175d40ec2f8d622a92ccac3e59419c773b4eaad2b5a2"
 function = "put"
 inputs = ["0x123", "map_contract"]
 
 [[call]]
 call_type = "deploy"
-class_hash = "0x5CADC0ECBC6E2A502A7A7E8E5B55400A1A92AFCCB68ED5DC5EFAD86C1FC4EDB"
+class_hash = "0x6e10d493d7c807e0fbaad4f0c31792f24d64747fa328830a68cb5d2313f9a55"
```
Comment (Arcticae) [inline crates/sncast/tests/data/multicall_configs/deploy_invoke_calldata_ids.toml:16]: why did this change the value?

**Review by Arcticae — CHANGES_REQUESTED** [review]
⬆️

File: crates/sncast/tests/data/multicall_configs/deploy_invoke_calldata_ids.toml (hunk 1 above)
Comment (THenry14) [reply crates/sncast/tests/data/multicall_configs/deploy_invoke_calldata_ids.toml:16]: The issue was left over, cached artifacts. This is something Igor is trying to fix in an issue he's working on

Comment (THenry14) [reply crates/sncast/tests/data/multicall_configs/deploy_invoke_calldata_ids.toml]: I have resolved the conflicts :+1:

**Review by Arcticae — APPROVED** [review]

**Pull request merged**